app/src/mutations/webhook.rs

```diff
@@ -126,6 +127,76 @@ impl Mutation {
             webhook: input.webhook,
         })
     }
+
+    /// Res
+    ///
+    /// # Errors
+    /// This function fails if ...
+    pub async fn edit_webhook(
+        &self,
+        ctx: &Context<'_>,
+        input: EditWebhookInput,
+    ) -> Result<EditWebhookPayload> {
+        let AppContext { db, user_id, .. } = ctx.data::<AppContext>()?;
+        let svix = ctx.data::<Svix>()?;
+
+        let user_id = user_id.ok_or_else(|| Error::new("X-USER-ID header not found"))?;
+
+        let webhook = webhooks::Entity::find()
+            .filter(webhooks::Column::Id.eq(input.webhook))
+            .one(db.get())
+            .await?
+            .ok_or_else(|| Error::new("webhook not found"))?;
+
+        let org_app = organization_applications::Entity::find()
+            .filter(organization_applications::Column::OrganizationId.eq(webhook.organization_id.clone()))
+            .one(db.get())
+            .await?
+            .ok_or_else(|| Error::new("organization not found"))?;
+
+        let app_id = org_app.svix_app_id;
+
+        let update_endpoint = EndpointUpdate {
+            channels: Some(input.projects.iter().map(ToString::to_string).collect()),
+            filter_types: Some(input.filter_types.iter().map(|e| e.format()).collect()),
+            version: 1,
+            description: Some(input.description),
+            disabled: input.disabled,
+            rate_limit: input.rate_limit,
+            url: input.endpoint,
+            uid: Some(webhook.id.clone().to_string()),
+        };
+
+        let endpoint = svix
+            .endpoint()
+            .update(app_id.clone(), webhook.endpoint_id.clone(), update_endpoint, None)
+            .await?;
+
+        for project in input.projects {
+            let webhook_project_active_model = webhook_projects::ActiveModel {
+                webhook_id: Set(webhook.id),
+                project_id: Set(project),
+                ..Default::default()
+            };
+
+            webhook_project_active_model.insert(db.get()).await?;
+        }
```

Comment:
Need to clear out the previous clear out the previous projects and write in the new ones from the edit inputs.

Can you see about doing both actions in a transaction so there is no time where the projects are completely cleared out.

Reply:
Sure